src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/CSharpDiagnosticWorkerWithAnalyzers.cs

```diff
@@ -183,6 +180,12 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs changeEv
                     var projectDocumentIds = _workspace.CurrentSolution.GetProject(changeEvent.ProjectId).Documents.Select(x => x.Id).ToImmutableArray();
                     QueueForAnalysis(projectDocumentIds, AnalyzerWorkType.Background);
                     break;
+                case WorkspaceChangeKind.SolutionAdded:
```

Comment:
this is the type of change that ideally should go in a separate PR - to make this one a bit more manageable. would it be a problem to extract it into a different one? thanks

Reply:
I think other changes except diagnostic api change (it affects many parts) are easy to extract in additional pr:s that can be merged before this change. There was lots of trial and error before this version and some refactorings originate from them even after they ended up as dead ends 😄 I will check what parts can be extracted out from this PR.